Common/src/matrix_structure.cpp

```diff
@@ -1457,327 +1049,35 @@ void CSysMatrix<ScalarType>::MatrixVectorProductTransposed(const CSysVector<Scal
-      aux_vector[jVar] = 0.0;
-    aux_vector[iVar] = 1.0;
-    
-    /*--- Compute the i-th column of the inverse matrix ---*/
-    
-    Gauss_Elimination_ILUMatrix(block_i, aux_vector);
-    for (jVar = 0; jVar < nVar; jVar++)
-      invBlock[jVar*nVar+iVar] = aux_vector[jVar];
-  }
-  
-  //  ScalarType Det, **Matrix, **CoFactor;
-  //  ScalarType *Block = GetBlock_ILUMatrix(block_i, block_i);
-  //
-  //  Matrix = new ScalarType*[nVar];
-  //  CoFactor = new ScalarType*[nVar];
-  //  for (iVar=0;iVar<nVar;iVar++) {
-  //    Matrix[iVar] = new ScalarType[nVar];
-  //    CoFactor[iVar] = new ScalarType[nVar];
-  //  }
-  //
-  //  for (iVar = 0; iVar < nVar; iVar++) {
-  //    for (jVar = 0; jVar < nVar; jVar++)
-  //    Matrix[iVar][jVar] = Block[jVar*nVar+iVar];
-  //  }
-  //
-  //  Det =  MatrixDeterminant(Matrix, nVar);
-  //  MatrixCoFactor(Matrix, nVar, CoFactor);
-  //  MatrixTranspose(CoFactor, nVar);
-  //
-  //
-  //  for (iVar = 0; iVar < nVar; iVar++) {
-  //    for (jVar = 0; jVar < nVar; jVar++)
-  //    invBlock[jVar*nVar+iVar] = CoFactor[iVar][jVar]/Det;
-  //  }
-  //
-  //  for (iVar = 0; iVar < nVar; iVar++) {
-  //    delete [] Matrix[iVar];
-  //    delete [] CoFactor[iVar];
-  //  }
-  //  delete [] Matrix;
-  //  delete [] CoFactor;
-  
-}
-
-template<class ScalarType>
-void CSysMatrix<ScalarType>::BuildJacobiPreconditioner(bool transpose) {
-
-  unsigned long iPoint, iVar, jVar;
-
-  /*--- Compute Jacobi Preconditioner ---*/
-  for (iPoint = 0; iPoint < nPoint; iPoint++) {
-
-    /*--- Compute the inverse of the diagonal block ---*/
-    InverseDiagonalBlock(iPoint, block_inverse, transpose);
-
-    /*--- Set the inverse of the matrix to the invM structure (which is a vector) ---*/
-    for (iVar = 0; iVar < nVar; iVar++)
-      for (jVar = 0; jVar < nVar; jVar++)
-        invM[iPoint*nVar*nVar+iVar*nVar+jVar] = block_inverse[iVar*nVar+jVar];
-  }
-
-}
-
-template<class ScalarType>
-void CSysMatrix<ScalarType>::ComputeJacobiPreconditioner(const CSysVector<ScalarType> & vec, CSysVector<ScalarType> & prod, CGeometry *geometry, CConfig *config) {
-  
-  unsigned long iPoint, iVar, jVar;
-  
-  for (iPoint = 0; iPoint < nPointDomain; iPoint++) {
-    for (iVar = 0; iVar < nVar; iVar++) {
-      prod[(unsigned long)(iPoint*nVar+iVar)] = 0.0;
-      for (jVar = 0; jVar < nVar; jVar++)
-        prod[(unsigned long)(iPoint*nVar+iVar)] +=
-        invM[(unsigned long)(iPoint*nVar*nVar+iVar*nVar+jVar)]*vec[(unsigned long)(iPoint*nVar+jVar)];
-    }
-  }
-  
-  /*--- MPI Parallelization ---*/
-  
-  InitiateComms(prod, geometry, config, SOLUTION_MATRIX);
-  CompleteComms(prod, geometry, config, SOLUTION_MATRIX);
-  
-}
-
-template<class ScalarType>
-unsigned long CSysMatrix<ScalarType>::Jacobi_Smoother(const CSysVector<ScalarType> & b, CSysVector<ScalarType> & x, CMatrixVectorProduct<ScalarType> & mat_vec, ScalarType tol, unsigned long m, ScalarType *residual, bool monitoring, CGeometry *geometry, CConfig *config) {
```

Comment:
Thanks for collapsing all of my old smoothers into a unified implementation. Can some of the comments here be preserved in the new implementations under the Compute*Preconditioner() routines? I recall adding many of them to explain step by step what each of the smoothers were doing...

Reply:
Sure can, I'll see if there were comments in the smoothers that are not in the preconditioner.